### PR TITLE
Only do runtime ownership of /app/media

### LINKED ISSUE #83: Make directory ownership change optional
We use `docker-compose` for development and map our source directory to the `/app` directory in the container. The container's `django-entrypoint.sh` contains a `chown -R django:django /app` which then affects the locally mounted directory. See:
https://github.com/praekeltfoundation/docker-django-bootstrap/blob/18ba534ca657992d46420cfbb1c283d6f0257524/django-entrypoint.sh#L17-L20

Would you consider adding an environment variable with which the changing of the ownership can be skipped?

### PR DESCRIPTION
**Pull request by JayH5**
Should fix #83

### CODE REVIEW

File: django-entrypoint.sh [hunk 1]
```diff
@@ -14,10 +14,11 @@ if [ "$1" = 'celery' ]; then
 fi
 
 if [ "$1" = 'gunicorn' ]; then
-  # Do an extra chown of the /app directory at runtime in addition to the one in
-  # the build process in case any directories are mounted as root-owned volumes
```
Comment (JayH5) [inline django-entrypoint.sh:18]: There hasn't been a chown during the build process for a while now. That was when we had an onbuild image.

**Comment by bearnard** [comment]
👍

File: django-entrypoint.sh [hunk 2]
```diff
@@ -14,10 +14,11 @@ if [ "$1" = 'celery' ]; then
...
-  # at runtime.
-  chown -R django:django /app
+  # Do a chown of the /app/media directory (if it exists) at runtime in case the
+  # directory was mounted as a root-owned volume.
+  if [ -d /app/media ] && [ "$(stat -c %u /app/media)" != "$(id -u django)" ]; then
```
Comment (cobusc) [inline django-entrypoint.sh]: Alternative: `[ "$(stat -c %U /app/media)" != "django" ]`

File: example/test.py [hunk 3]
```diff
@@ -415,6 +415,16 @@ def test_gzip_small_file_not_compressed(self, web_container):
             Not(Contains('Vary')),
         ))
 
+    def test_media_volume_ownership(self, web_container):
+        """
+        The /app/media directory should have the correct ownership set at
+        runtime if it is a mounted volume.
+        """
+        [app_media_ownership] = web_container.exec_run(
+            ['stat', '-c', '%u:%g', '/app/media'])
```
Comment (cobusc) [inline example/test.py]: If using "%U:%G" you can check actual user and group names in the assert below.

**Comment by cobusc** [comment]
:+1:

**Pull request merged**